torch/_dynamo/variables/builtin.py

```diff
@@ -846,12 +846,24 @@ def handle_is(tx: "InstructionTranslator", left, right):
                         return ConstantVariable.create(op.__name__ != "is_")
                     if left is right:
                         return ConstantVariable.create(op(left, right))
-                    if (
-                        istype(left, variables.ExceptionVariable)
-                        and istype(right, variables.ExceptionVariable)
-                        and left.exc_type is not right.exc_type
+                    if istype(left, variables.ExceptionVariable) and istype(
+                        right, variables.ExceptionVariable
                     ):
-                        return ConstantVariable.create(op(left, right))
+                        if (left.exc_type is not right.exc_type) or (
+                            len(left.args) != len(right.args)
+                        ):
+                            return ConstantVariable.create(
+                                False if op is operator.is_ else True
+                            )
+                        # They cannot be the same object if the arguments are different
+                        m = BuiltinVariable(operator.eq).call_function
+                        if any(
+                            not m(tx, [a, b], {}).value  # type: ignore[attr-defined]
+                            for a, b in zip(left.args, right.args)
```

Comment:
er, what happens if there are different number of args

Reply:
Changed the code to consider this as well